src/renderer/Material.ts

```diff
@@ -0,0 +1,66 @@
+import { vec3 } from 'gl-matrix';
+import { Bakeable, Color } from '../calder';
+import { RGBColor } from '../colors/RGBColor';
+
+/**
+ * Required parameters when defining a `Material`.
+ */
+export type MaterialParams = {
+    color: Color;
+    shininess: number;
+};
+
+/**
+ * The representation of a `Material` when passed to the shader.
+ */
+export type BakedMaterial = {
+    materialColor: vec3;
+    materialShininess: number;
+};
+
+/**
+ * A simple representation of the material of an object in a scene.
+ */
+export class Material implements Bakeable {
+    public readonly materialColor: vec3;
+    public readonly materialShininess: number;
+
+    /**
+     * Instantiates a new Material.
+     *
+     * @class Material
+     * @constructor
+     * @param {MaterialParams} params The named parameters for this material.
+     */
+    constructor(params: MaterialParams) {
```

Comment:
Should we make this constructor `private` due to the `public static create` method below?